### PR TITLE
live.py: harmonize xorrisofs arguments with lorax live/x86.tmpl

### PR DESCRIPTION
**Pull request by FGrose**
```code
Reorder xorrisofs options and arguments to match those in the
lorax live/x86.tmpl template because they receive more quality
control testing.
  For the -e images/efiboot.img section,
    remove -isohybrid-apm-hfsplus.
  For the -e images/macboot.img section,
    remove -isohybrid-gpt-basdat & -isohybrid-apm-hfsplus,
    add -isohybrid-gpt-hfsplus.
    Make this section conditional on the existence of
    images/macboot.img.
  Remove option -hide-rr-moved.

Unlike lorax, invoke the -iso-level 3 option based on the maximum
  individual file size in the isodir, as should be.
```

### CODE REVIEW

File: imgcreate/live.py [hunk 1]
```diff
@@ -320,10 +320,23 @@ def __create_iso(self, isodir):
...
+        def iterate_files_recursively(directory="."):
+            for entry in os.scandir(directory):
+                if entry.is_dir():
+                    for file in iterate_files_recursively(entry.path):
+                        yield file
+                else:
+                    yield entry.stat().st_size
+
+        if max(iterate_files_recursively(isodir)) >= 4 * 1024**3:
+            [args.insert(1, i) for i in ['3', '-iso-level']]
```
Comment (bcl) [inline imgcreate/live.py]: That's kind of an odd way to add the argument? Why did you do it that way?

Comment (FGrose) [reply imgcreate/live.py]: I like the concise form of a [list comprehension](http://www.siafoo.net/article/52#id18). Reversing the insertion list makes it easy to match the order desired.

Comment (FGrose) [reply imgcreate/live.py]: Sorry, I force pushed a more traditional approach before seeing that you may have offered a change (that is now outdated).

Comment (bcl) [reply imgcreate/live.py]: Nope, I hadn't added any other comments. I'm a fan of list comprehension as well, but that line is *really* hard to understand on first or second look :) The updated version is much easier to read.

**Pull request merged**